plugin/pkg/scheduler/factory/plugins.go

```diff
@@ -356,9 +358,27 @@ func getPriorityFunctionConfigs(names sets.String, args PluginFactoryArgs) ([]al
 			})
 		}
 	}
+	var totalPriority int
+	for _, config := range configs {
+		// Checks for overflow and returns errors.
+		// TODO: Need to check overflow for weight*10.
+		if checkOverFlow(totalPriority, config.Weight*10) {
+			return nil, fmt.Errorf("Total priority of priority functions has overflown")
+		}
+		totalPriority += config.Weight * 10
+	}
 	return configs, nil
 }
 
+// checkOverFlow checks if totalPriority can overflow after adding it with weight.
+func checkOverFlow(totalPriority int, weight int) bool {
+	if totalPriority == MaxInt || weight == MaxInt {
+		return true
+	}
+	sum := totalPriority + weight
```

Comment:
In Go, I'm not sure whether `sum` is still a valid value after overflow; in C, the `sum` maybe randoms after overflow (e.g. checking by interrupt or others) in some arch. 

So I'd prefer to 

```
return totalPriority >= MaxInt || weight >= MaxInt || weight >= MAX_INT - totalPriority
```